torchtitan/protocols/hf_adapter.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from abc import ABC, abstractmethod
+from typing import Any
+
+import torch.nn as nn
+
+
+class HFAdapter(ABC):
+    """Abstract base class for HuggingFace state dict transformations.
+
+    This class defines the interface for converting between native model
+    state dict format and HuggingFace model state dict format.
+    """
+
+    @staticmethod
+    @abstractmethod
+    def get_hf_state_dict(state_dict: dict[str, Any]) -> dict[str, Any]:
+        """Convert from native model state dict to HuggingFace format.
+
+        Args:
+            state_dict: The native model state dict
+
+        Returns:
+            The converted HuggingFace format state dict
+        """
+        pass
+
+    @staticmethod
+    @abstractmethod
+    def set_from_hf_state_dict(model: nn.Module, hf_state_dict: dict[str, Any]) -> None:
```

Comment:
`convert_form_hf_state_dict`? Unlike `convert_to_hf_state_dict`, I don't know if `conver_from_hf_state_dict` is a good name, another name I prefer is `convert_to_titan_state_dict`.